### PR TITLE
ARTEMIS-3538 fix extra-tests formatting and examples compilation

### PR DESCRIPTION
**Pull request by tlbueno**
- Checkstyle verification fails on JMSXDeliveryCountTest
because of empty line with trailing spaces
```
[INFO] --- maven-checkstyle-plugin:3.1.1:check (default) @ extra-tests ---
[INFO] Starting audit...
[ERROR] /home/buildadm/workspace/upstream-test-suite/validate-install-tests/validate-install-tests-upstream/artemis-src/tests/extra-tests/src/test/java/org/apache/activemq/artemis/tests/extras/jms/xa/JMSXDeliveryCountTest.java:681: Line has trailing spaces. [RegexpSingleline]
Audit done.
[INFO] There is 1 error reported by Checkstyle 8.43 with /home/buildadm/workspace/upstream-test-suite/validate-install-tests/validate-install-tests-upstream/artemis-src/tests/extra-tests/../../etc/checkstyle.xml ruleset.
[ERROR] src/test/java/org/apache/activemq/artemis/tests/extras/jms/xa/JMSXDeliveryCountTest.java:[681] (regexp) RegexpSingleline: Line has trailing spaces.
```

- examples classes DummyXid failed to compile because of
uncommented separation comment:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project xa-heuristic: Compilation failure: Compilation failure: 
[ERROR] /home/tbueno/repos/github.com/tlbueno/activemq-artemis/examples/features/standard/xa-heuristic/src/main/java/org/apache/activemq/artemis/jms/example/DummyXid.java:[101,1] illegal start of type
[ERROR] /home/tbueno/repos/github.com/tlbueno/activemq-artemis/examples/features/standard/xa-heuristic/src/main/java/org/apache/activemq/artemis/jms/example/DummyXid.java:[101,3] ';' expected
[ERROR] /home/tbueno/repos/github.com/tlbueno/activemq-artemis/examples/features/standard/xa-heuristic/src/main/java/org/apache/activemq/artemis/jms/example/DummyXid.java:[101,5] illegal start of type
[ERROR] /home/tbueno/repos/github.com/tlbueno/activemq-artemis/examples/features/standard/xa-heuristic/src/main/java/org/apache/activemq/artemis/jms/example/DummyXid.java:[101,7] <identifier> expected
[ERROR] /home/tbueno/repos/github.com/tlbueno/activemq-artemis/examples/features/standard/xa-heuristic/src/main/java/org/apache/activemq/artemis/jms/example/DummyXid.java:[101,9] ';' expected
```

### CODE REVIEW

**Comment by brusdev** [comment]
@tlbueno please replace NO-JIRA with ARTEMIS-3538

**Review by brusdev — APPROVED** [review]

**Comment by gemmellr** [comment]
Was about to push a fix but hadnt seen this opened, I'll leave it to @brusdev to apply this.

**Comment by brusdev** [comment]
@gemmellr I'll merge this PR soon after checks

**Pull request merged**
